test/metabase/models/setting_test.clj

```diff
@@ -15,10 +17,14 @@
    [metabase.util :as u]
    [metabase.util.encryption-test :as encryption-test]
    [metabase.util.i18n :as i18n :refer [deferred-tru]]
+   [metabase.util.log :as log]
    [toucan2.core :as t2]))
 
 (use-fixtures :once (fixtures/initialize :db))
 
+;; TODO: it would be nice to detect and discard the changes to *all* settings defined in this namespace
+(use-fixtures :each (fn [thunk] (mt/discard-setting-changes [:test-setting-custom-init] (thunk))))
```

Comment:
`use-fixtures` is used sparingly for a good reason: it's difficult to notice it's happening when reading the test in isolation. This fixture is only helpful for a few tests, so IMO it doesn't cross the threshold. It's okay and preferable to duplicate this many times.

Reply:
As stated in my comment, I see this fixture as becoming a "hygiene for internal settings" block that applies to all the tests. With the current helpers that would have two big issues:

1. All the settings would need to be explicitly enumerated for the macro.
2. All the settings would be captured and re-set, even if they weren't touched.

Even with a single setting, repeating the block across 4 tests feels brittle to me, but I'll go with your suggestion for now. Gets rid of a TODO 😆